### DISCUSSION TITLE
unocss vscode 的插件提供的代码提示能够给出更多选择吗，类似 WindiCSS IntelliSense

### DISCUSSION THREAD

**Discussion by xby020** [unocss/unocss#1776] [Ideas]
在输入代码时，一般需要输入一项属性的大部分值才会提供补全，像`overflow-hidden`, 在输入'overflow"之后才会提供选项：
<img width="387" alt="image" src="https://user-images.githubusercontent.com/26197465/196636823-6fce2599-b66d-4063-8c5b-63ced1df5dd7.png">

WindiCSS IntelliSence 则能够提供更多选项：
<img width="341" alt="image" src="https://user-images.githubusercontent.com/26197465/196637760-b4ba2968-a2ed-4792-8370-8038b9657a1f.png">

在输入补全时能够补全中间输入缺失的属性：
<img width="245" alt="image" src="https://user-images.githubusercontent.com/26197465/196638165-d13f68bd-8760-443e-bd5d-a3c02c83800a.png">

希望能够增强补全功能，敲代码更丝滑~

**Comment by zyyv** [comment]
For some static rules, UnoCSS can quickly provide your autocomplete template; dynamic rules may be slower, depending on what template the rule is configured with.
You can also add your custom autocomplete templates in `Configuration` or `RuleMeta`.

**Reply by xby020** [reply]
在未来是否有可能将不同的代码自动补全片段通过预设或者插件设置提供？
Is it possible in the future to provide different snippets of code autocomplete via preset package or extension settings?

**Reply by zyyv** [reply]
我想我们已经提供了，规则一般来自于预设，静态规则已经被收纳到自动填充模版了，而动态规则或`shortcuts`在第三个参数中`autocomplete`配置模板，现有的规则应该都配置好了。如果你觉得`presetMini` `presetWind`的填充模板配置有遗漏或者不够完善，你可以提PR进行完善。😄
I think we have already provided it. The rules are generally from presets, `static rules` have been included in the autocomplete template, and dynamic `rules` or `shortcuts` in the third parameter `autocomplete` configuration template, the existing rules should all be Configured. If you feel that the `presetMini` `presetWind` fill template configuration is missing or not perfect, you can submit a PR for improvement. 😄
